### PR TITLE
Add ORB float32 support

### LINKED ISSUE #4675: ORB detection doesn't work on 32 bit float input
## Description

Trying to use ORB feature detection of an image of dtype 'f4' fails, as the function [_corner_fast](https://github.com/scikit-image/scikit-image/blob/master/skimage/feature/corner_cy.pyx#L110) requires `double` input, but nothing converts it to double on the way. Integer input works however.

## Way to reproduce
```python
from skimage import data, feature

img = data.checkerboard().astype('f4')
orb = feature.ORB()
orb.detect(img)
```

## Version information

```python
3.8.2 (default, Feb 28 2020, 00:00:00) 
[GCC 10.0.1 20200216 (Red Hat 10.0.1-0.8)]
Linux-5.6.8-300.fc32.x86_64-x86_64-with-glibc2.2.5
scikit-image version: 0.17.1
numpy version: 1.18.4

```

### PR DESCRIPTION
**Pull request by rfezzani**
## Description

Fixes #4675.

## Checklist

- [ ] [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- [ ] Gallery example in `./doc/examples` (new features only)
- [ ] Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- [ ] Unit tests
- [ ] Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)

## For reviewers

- [ ] Check that the PR title is short, concise, and will make sense 1 year
  later.
- [ ] Check that new functions are imported in corresponding `__init__.py`.
- [ ] Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.

### CODE REVIEW

File: skimage/feature/tests/test_orb.py [hunk 1]
```diff
@@ -120,3 +122,10 @@ def test_no_descriptors_extracted_orb():
     detector_extractor = ORB()
     with testing.raises(RuntimeError):
         detector_extractor.detect_and_extract(img)
+
+
+@pytest.mark.parametrize('dtype', ['float32', 'float64', 'uint8', 'int'])
```
Comment (hmaarrfk) [inline skimage/feature/tests/test_orb.py:127]: Int64?

Comment (rfezzani) [reply skimage/feature/tests/test_orb.py:127]: :+1: I am trying to find a more elaborated test

Comment (emmanuelle) [reply skimage/feature/tests/test_orb.py:127]: let me still merge this now you're welcome to submit an other PR with a more complete test of course.

**Comment by emmanuelle** [comment]
@meeseeksdev backport to v0.17.x

**Comment by rfezzani** [comment]
@emmanuelle, I found that `orb_loop` only supports `double`, I will come with a PR for fixing this also soon

**Pull request merged**